provisioner/aws-manager.js

```diff
@@ -15,9 +17,12 @@ function AwsManager(ec2, keyPrefix, pubKey) {
   this.ec2 = ec2;
   this.keyPrefix = keyPrefix;
   this.pubKey = pubKey;
-  this.__apiState = {};
-  this.__internalState = {};
   this.__knownKeyPairs = [];
+  this.__apiState = {
+    instances: [],
+    requests: [],
+  };
+  this.__internalState = [];
```

Comment:
My line of thinking is that I have __apiState and __internalState.  __submittedSpotReq to me doesn't make it any more clear.  I think adding a comment here is probably the better option.